### PR TITLE
[DMS-74] Implement missing functionality for set

### PR DESCRIPTION
**Pull request by Sereja313**
Add following functions:

union(rbSet1 : Set<T>, rbSet2 : Set<T>) : Set<T>
intersect(rbSet1 : Set<T>, rbSet2 : Set<T>) : Set<T>
diff(rbSet1 : Set<T>, rbSet2 : Set<T>) : Set<T>
map<T1>(rbSet : Set<T1>, f : T1 -> T) : Set<T>
mapFilter<T1>(rbSet: Set<T1>, f : T1 -> ?T) : Set<T>
isSubset(rbSet1 : Set<T>, rbSet2 : Set<T>) : Bool
equals (rbSet1 : Set<T>, rbSet2 : Set<T>) : Bool
isEmpty<T> (rbSet : Set<T>) : Bool

Add tests for the functions above.

### CODE REVIEW

File: src/PersistentOrderedMap.mo [hunk 1]
```diff
@@ -574,6 +574,52 @@ module {
     acc
   };
 
+  /// Helper function, joins an element and two trees.
```
Comment (s-and-witch) [inline src/PersistentOrderedMap.mo]: I don't feel very comfortable that we import helper functions as part of our public API.

We may move these functions inside `Internal` module and make it public, so you can call them as `Map.Internal.join`.

**Review by s-and-witch — COMMENTED** [review]
Overall looks fine to me, just one comment

File: src/PersistentOrderedMap.mo (hunk 1 above)
Comment (Sereja313) [reply src/PersistentOrderedMap.mo]: Yeah, that makes sense, fixed

File: src/PersistentOrderedSet.mo [hunk 2]
```diff
@@ -4,6 +4,9 @@
 /// * Runtime: `O(log(n))` worst case cost per insertion, removal, and retrieval operation.
 /// * Space: `O(n)` for storing the entire tree.
 /// `n` denotes the number of elements (i.e. nodes) stored in the tree.
+///
+/// The set operations iplemntation is derived from:
```
Comment (GoPavel) [inline src/PersistentOrderedSet.mo]:
```suggestion
/// The set operations implementation is derived from:
```

**Review by GoPavel — APPROVED** [review]
LGTM

**Pull request merged**